src/reactionsystem.jl

```diff
@@ -1388,7 +1467,8 @@ function MT.flatten(rs::ReactionSystem; name = nameof(rs))
         balanced_bc_check = false,
         spatial_ivs = get_sivs(rs),
         continuous_events = MT.continuous_events(rs),
-        discrete_events = MT.discrete_events(rs))
+        discrete_events = MT.discrete_events(rs),
+        metadata = MT.get_metadata(rs))
```

Comment:
Bugfix. Currently, flattening a system removes any system metadata.

Reply:
Agreed this is a bit of a mess, but I think this replicated the behaviour MTK has (i.e. flattening used to remove metadata, and now it doesn't anymore).

In practise the main problem is that I have never really got a reply of what the `metadata` field actually is. I use it a bit for a personal project. But if we can confirm whether it is for users benefit, or if MTK at some point want to stick stuff here, that changes things a lot.

Reply:
Sounds good.